### PR TITLE
Feat: Remove isCurrentUserController dependency

### PR DESCRIPTION
**Pull request by lmuntanerDfinity**
# Motivation

We are going to remove the property isCurrentUserController from the Neuron type in nns-js. We first need to remove the dependency here.

# Changes

* Neuron util `isCurrentUserController` now checks that the controller is the main account's principal.

# Tests

* Update tests to new functionality.

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
@mstrasinskis @peterpeterparker after discussing the @ielashi we should remove the property added by nns-js `isCurrentUserController` and instead, each app should do they own logic for it, not nns-js.

This PR is the first step towards removing it from nns-js.

**Comment by lmuntanerDfinity** [comment]
@peterpeterparker @mstrasinskis please review

File: frontend/svelte/src/lib/utils/neuron.utils.ts [hunk 1]
```diff
@@ -90,10 +90,13 @@ export const formatVotingPower = (value: bigint): string =>
 export const hasJoinedCommunityFund = (neuron: NeuronInfo): boolean =>
   neuron.joinedCommunityFundTimestampSeconds !== undefined;
 
-export const isCurrentUserController = (neuron: NeuronInfo): boolean =>
-  neuron.fullNeuron?.isCurrentUserController === undefined
+export const isCurrentUserController = (
+  neuron: NeuronInfo,
+  accounts: AccountsStore
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/utils/neuron.utils.ts:95]: - why passing the all store and not just the main account?

- nth: object params `{neuron, mainAccount}`

**Comment by peterpeterparker** [comment]
> We first need to remove the dependency here.

Why? The nns-js dependency is pinned we can proceed with both in parallel no?

File: frontend/svelte/src/lib/utils/neuron.utils.ts (hunk 1 above)
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/utils/neuron.utils.ts:95]: sure, I wanted to keep the same signature of the other helper, but it also works passing the main only

**Comment by lmuntanerDfinity** [comment]
Well, just in case I change the nns-js, then we need to update nns-js because someone else adds functionality there that we need.

But yes, regarding only this refactor, we can proceed in parallel.

**Review by peterpeterparker — APPROVED** [review]

**Pull request merged**